keysign/bluetoothoffer.py

```diff
@@ -76,6 +76,10 @@ def start(self):
         returnValue((success, message))
 
     def allocate_code(self):
+        # Do not allocate the BT code if there isn't a powered on BT adapter
+        if not get_bt_power():
+            self.code = None
+            return None
```

Comment:
ouf. confusing.
can you make it return only once in this method?

also, does this method always set `self.code` to something? Then it may be smart to also do that just before returning.